crates/node/gossip/src/driver.rs

```diff
@@ -196,16 +196,18 @@ where
                 {
                     if id == listener_id {
                         info!(target: "gossip", "Swarm now listening on: {address}");
-                        break;
+
+                        self.addr = address.clone();
+
+                        return Ok(address);
```

Comment:
I don't think we need the return here since the `Ok(())` was removed. Should be able to just be:
```suggestion
                        Ok(address)
```